cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/CgmesExportUtil.java

```diff
@@ -338,4 +338,33 @@ public static boolean isEquivalentShuntWithZeroSectionCount(Connectable<?> c) {
         }
         return false;
     }
+
+    public static <I extends ReactiveLimitsHolder & Injection<I>> ReactiveCapabilityCurve obtainCurve(I i) {
+        return i.getReactiveLimits().getKind().equals(ReactiveLimitsKind.CURVE) ? i.getReactiveLimits(ReactiveCapabilityCurve.class) : null;
+    }
+
+    public static String obtainCalculatedSynchronousMachineKind(double minP, double maxP, ReactiveCapabilityCurve curve) {
+        double min = curve != null ? curve.getMinP() : minP;
+        double max = curve != null ? curve.getMaxP() : maxP;
+
+        String kind;
+        if (min > 0) {
+            kind = "generator";
+        } else if (max < 0) {
+            kind = "motor";
+        } else if (min == 0 && max == 0) {
+            kind = "condenser";
+        } else if (min == 0) {
+            kind = "generatorOrCondenser";
+        } else if (max == 0) {
+            kind = "motorOrCondenser";
+        } else {
+            kind = "generatorOrCondenserOrMotor";
+        }
+        return kind;
```

Comment:
In case of new model creation (CGMES data not imported), it seems that `generatorOrMotor` kind is not created.
Is that information acquired from IIDM in this case or it's not exported?

Reply:
With the available information in IIDM Model, there is no way to discriminate the `generatorOrMotor` kind.

Reply:
@annetill Can we discuss `generatorOrMotor` topic?
`generatorOrCondenserOrMotor` is generally rare type of machine to be applied to all machines which can act as motor and generator.

We mostly have `generatorOrMotor` for reversible hydro machines.